lib/Interfaces.ts

```diff
@@ -22,7 +22,10 @@ export interface IRequestOptions {
     proxy?: IProxyConfiguration,
     cert?: ICertConfiguration,
     allowRedirects?: boolean, 
-    maxRedirects?: number
+    maxRedirects?: number,
+    reuseAgent? : boolean,
+    maxSockets?: number,
+    keepAlive?: boolean
```

Comment:
After talking here, I think we just want a keepAlive option and if true, we reuse agents / connections.  There's no scenarios where I want keepAlive but not reuse.

Since keepAlive defaults to false in node let's default to false.

If you're using the rest client, I think this means you need to add that option to the rest client and it passes it through.

Reply:
done using keepAlive now. We aren't using rest client, but it looks like rest client is sharing the same interface (https://github.com/Microsoft/typed-rest-client/blob/master/lib/RestClient.ts#L37) so these options should work for rest client too.